Assets/Global/Scripts/UIScripts/Healthbar.cs

```diff
@@ -14,18 +14,17 @@ void Awake()
         HealthbarOverlayTransform = transform.GetChild(1) as RectTransform;
 
         originalWidth = HealthbarUnderlayTransform.sizeDelta.x;
+        GlobalReference.SubscribeTo(Events.HEALTH_CHANGED, UpdateCurrentHealthPermanent);
```

Comment:
Can you explain what why you are changing the permanent stats now?
I have a hard time following what your thought process was here. 

Note again that the Permanent stats are only modifiers and should never be changed.
So in theory if the code was all implemented, it should not have a change in the Healthbar at all